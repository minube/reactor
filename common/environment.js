import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
