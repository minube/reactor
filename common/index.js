import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
