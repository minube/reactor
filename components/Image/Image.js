import React, { PureComponent } from 'react';
import { Image as ImageNative, StyleSheet, View } from 'react-native';

import Activity from '../Activity';
import styles from './Image.style';

class Image extends PureComponent {
  state = {
    ready: false,
  }

  _onLoad = (event) => {
    const { props: { ...inherit } } = this;

    this.setState({ ready: true });
    if (inherit.onLoad) inherit.onLoad(event);
  }

  render() {
    const {
      _onLoad,
      props: { ...inherit },
      state: { ready },
    } = this;

    return (
      <View style={StyleSheet.flatten([styles.container, inherit.style])}>
        <ImageNative {...inherit} onLoad={_onLoad} />
        { !ready && <Activity color="white" size={inherit.size || 'large'} style={styles.activity} /> }
      </View>
    );
  }
}

Image.propTypes = {};

Image.defaultProps = {};

export default Image;
