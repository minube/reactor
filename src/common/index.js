import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
