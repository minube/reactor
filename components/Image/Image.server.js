import { createElement } from 'react';

import { serverSideStyle } from '../../common';
import resizeWithAkamai from './modules/resizeWithAkamai';

const DEFAULT_ALT = 'minube.com';

export default ({
  alt = DEFAULT_ALT, source: { uri }, style, title,
}) => createElement('img', {
  alt,
  src: resizeWithAkamai({ uri }),
  style: serverSideStyle(style),
  title,
});
