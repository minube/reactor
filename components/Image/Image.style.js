import { StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';

const { COLOR } = THEME;

export default StyleSheet.create({
  container: {
    ...LAYOUT.STYLE.CENTERED,
    backgroundColor: COLOR.BACKGROUND,
  },

  skeleton: {
    position: 'absolute',
    left: 0,
    top: 0,
    width: '100%',
    height: '100%',
  },
});
