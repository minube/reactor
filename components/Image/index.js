import { ENV } from '../../common';
import Default from './Image';

const Loading = ENV.IS_SERVER ? require('./Image.server').default : Default;

export default Loading;
