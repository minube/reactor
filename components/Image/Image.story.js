import React from 'react';
import { storiesOf } from '@storybook/react';

import Image from './Image';

const IMAGE_PLACEHOLDER = 'https://picsum.photos/320/200/?random';

storiesOf('âœ… Image', module)
  .addWithJSX('default', () => (
    <Image resizeMode="cover" source={{ uri: IMAGE_PLACEHOLDER }} style={{ width: 640, height: 400 }} />
  ))
  .addWithJSX('without source', () => (
    <Image style={{ width: 320, height: 240 }} />
  ))
  .addWithJSX('ðŸ’„ Custom Style', () => (
    <Image
      source={{ uri: IMAGE_PLACEHOLDER }}
      style={{
        borderRadius: 64,
        height: 128,
        width: 128,
      }}
    />
  ));
