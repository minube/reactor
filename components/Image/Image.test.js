import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Image from './Image';

const dataSource = [
  { caption: 'Spain', href: '/' },
  { caption: 'Basque Country', href: '/' },
  { caption: 'Bilbao', href: '/' },
];

const IMAGE_PLACEHOLDER = 'https://picsum.photos/320/200/?random';
const IMAGE_AKAMAI = 'https://imgs-akamai.mnstatic.com//tools//genericimages//fc5b42a5baceed399c3bb913406693fd3d4d64ef8bb69457697c90cb71b2c4dd.jpg';

describe('<Image>', () => {
  it('renders', () => {
    const tree = renderer.create(<Image />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {source}', () => {
    const tree = renderer.create(<Image source={{ uri: IMAGE_PLACEHOLDER }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {responsive}', () => {
    const tree = renderer.create(<Image responsive source={{ uri: IMAGE_AKAMAI }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {ratio}', () => {
    const tree = renderer.create(<Image ratio={10} responsive source={{ uri: IMAGE_AKAMAI }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color} (only Native)', () => {
    const tree = renderer.create(<Image color={THEME.COLOR.ERROR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {invert} (only Web)', () => {
    const tree = renderer.create(<Image invert />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {size}', () => {
    const tree = renderer.create(<Image size={48} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Image style={{ backgroundColor: THEME.COLOR.ACCENT }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Image />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

