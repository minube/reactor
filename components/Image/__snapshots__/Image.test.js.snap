// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image> renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#F5F5F6",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    onLoad={[Function]}
    source={undefined}
  />
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "position": "absolute",
        },
        false,
      ]
    }
  />
</View>
`;

exports[`<Image> when {color} (only Native) 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#F5F5F6",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    color="#d33c44"
    onLoad={[Function]}
    source={undefined}
  />
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "position": "absolute",
        },
        false,
      ]
    }
  />
</View>
`;

exports[`<Image> when {invert} (only Web) 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#F5F5F6",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    invert={true}
    onLoad={[Function]}
    source={undefined}
  />
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "position": "absolute",
        },
        false,
      ]
    }
  />
</View>
`;

exports[`<Image> when {ratio} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#F5F5F6",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    onLoad={[Function]}
    source={
      Object {
        "uri": "https://imgs-akamai.mnstatic.com//tools//genericimages//fc5b42a5baceed399c3bb913406693fd3d4d64ef8bb69457697c90cb71b2c4dd.jpg?fit=around|102:76",
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "position": "absolute",
        },
        false,
      ]
    }
  />
</View>
`;

exports[`<Image> when {responsive} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#F5F5F6",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    onLoad={[Function]}
    source={
      Object {
        "uri": "https://imgs-akamai.mnstatic.com//tools//genericimages//fc5b42a5baceed399c3bb913406693fd3d4d64ef8bb69457697c90cb71b2c4dd.jpg?fit=around|1024:768",
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "position": "absolute",
        },
        false,
      ]
    }
  />
</View>
`;

exports[`<Image> when {size} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#F5F5F6",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    onLoad={[Function]}
    size={48}
    source={undefined}
  />
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size={48}
    style={
      Array [
        Object {
          "position": "absolute",
        },
        false,
      ]
    }
  />
</View>
`;

exports[`<Image> when {source} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#F5F5F6",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    onLoad={[Function]}
    source={
      Object {
        "uri": "https://picsum.photos/320/200/?random",
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "position": "absolute",
        },
        false,
      ]
    }
  />
</View>
`;

exports[`<Image> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#F5F5F6",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#a66ecc",
      },
    ]
  }
>
  <Image
    onLoad={[Function]}
    source={undefined}
    style={
      Object {
        "backgroundColor": "#a66ecc",
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "position": "absolute",
        },
        false,
      ]
    }
  />
</View>
`;
